Carbon.Core/Functions/ConvertTo-CBase64.ps1

```diff
@@ -0,0 +1,198 @@
+
+function ConvertTo-CBase64
+{
+    <#
+    .SYNOPSIS
+    Base64 encodes things.
+    
+    .DESCRIPTION
+    The `ConvertTo-CBase64` function base64 encodes things. Pipe what you want to encode to `ConvertTo-CBase64`. The
+    function can encode:
+    
+    * [String]
+    * [byte]
+    * [char]
+    * Signed integers: [int16], [int], [int64] (i.e. [long])
+    * Unsigned integers: [uint16], [uint32], [uint64]
+    * Floating point numbers: [float], [double]
+    * [bool]
+
+    For each item piped to `ConvertTo-CBase64`, the function returns that item base64 encoded.
+
+    If you pipe all bytes or all chars to `ConvertTo-CBase64`, it will encode all the bytes and chars together. This
+    allows you to do this:
+
+        [IO.File]::ReadAllBytes('some file') | ConvertTo-CBase64
+
+    and get back a single string for all the bytes/chars.
+
+    By default, `ConvertTo-CBase64` uses Unicode/UTF-16 encoding when converting strings to base64 (this is the default
+    encoding of strings by .NET and PowerShell). To use a different encoding, pass it to the `Encoding` parameter
+    (`[Text.Encoding] | Get-Member -Static` will show all the default encodings).
+
+    .EXAMPLE
+    'Encode me, please!' | ConvertTo-CBase64
+    
+    Demonstrates how to encode a string in base64.
+    
+    .EXAMPLE
+    'Encode me, please!' | ConvertTo-CBase64 -Encoding ([Text.Encoding]::ASCII)
+    
+    Demonstrates how to use a custom encoding when converting a string to base64. The parenthesis around the encoding
+    is required by the PowerShell language.
+
+    .EXAMPLE
+    [IO.File]::ReadAllBytes('path to some file') | ConvertTo-CBase64
+
+    Demonstrates that you can pipe an array of bytes to `ConvertTo-CBase64` and you'll get back a single string of all
+    the bytes base64 encoded.
+
+    .EXAMPLE
+    [IO.File]::ReadAllText('path to some file').ToCharArray() | ConvertTo-CBase64
+
+    Demonstrates that you can pipe an array of chars to `ConvertTo-CBase64` and you'll get back a single string of all
+    the chars base64 encoded.
+
+    .EXAMPLE
+    @( $true, [int16]1, [int]2, [long]3, [uint16]4, [uint32]5, [uint64]6, [float]7.8, [double]9.0) | ConvertTo-CBase64
+
+    Demonstrates that `ConvertTo-CBase64` can convert booleans, all sizes of signed and unsigned ints, floats, and 
+    doubles to base64.
+    #>
+    [CmdletBinding()]
+    [OutputType([String])]
+    param(
+        [Parameter(Mandatory,ValueFromPipeline,Position=0)]
+        [AllowNull()]
+        [AllowEmptyString()]
+        # The value to base64 encode.
+        [Object]$InputObject,
+        
+        # The encoding to use. Default is Unicode/UTF-16 (the default .NET encoding for strings). This parameter is only
+        # used if encoding a string or char array.
+        [Text.Encoding]$Encoding = ([Text.Encoding]::Unicode)

```

Comment:
Not according to their docs:

![image](https://user-images.githubusercontent.com/1082153/106217346-e7110c00-6189-11eb-8d55-d38f3ce82fa6.png)

I checked .NET 2.1 and 3.1 and they all say that strings are UTF-16.

I looked at `[Text.Encoding]::Default`, and it is the operating system's default encoding, not .NET's internal string object encoding.

Reply:
> I looked at [Text.Encoding]::Default, and it is the operating system's default encoding, not .NET's internal string object encoding.

That can't be right because it's different for Windows PowerShell (`Windows-1252`) and PowerShell Core on Windows (`utf-8`).

But yeah you're right, looks like the internal implementation of strings in .NET is still UTF-16. And what `[Text.Encoding]::Default` actually does is define the default encoding that's used when .NET is reading/writing text to and from memory. .NET Core/5 will by default decode input text as UTF-8 and encode it as UTF-16 in memory, and vice versa.